### ISSUE TITLE
Test failure baseservices/threading/generics/TimerCallback/thread21/thread21.sh

### ISSUE THREAD

**Issue by VincentBu** [dotnet/runtime#50762]
Run: [runtime-coreclr gcstress-extra 20210404.2](https://dev.azure.com/dnceng/public/_build/results?buildId=1072315&view=ms.vss-test-web.build-test-results-tab&runId=32938970&resultId=102483&paneView=debug)

Failed tests:
```
CoreCLR OSX arm64 Checked gcstress0xc_jitstress1 @ OSX.1100.ARM64.Open
 -baseservices/threading/generics/TimerCallback/thread21/thread21.sh
 -baseservices/threading/generics/TimerCallback/thread19/thread19.sh
```

**Error message:**
```
Assert failure(PID 46534 [0x0000b5c6], Thread: 20928983 [0x13f59d7]): (pMDReal != NULL) || !pCF->IsFrameless()
 File: /Users/runner/work/1/s/src/coreclr/vm/gcenv.ee.common.cpp Line: 370
 Image: /private/tmp/helix/working/9BD708E3/p/corerun

dyld: Library not loaded: @rpath/libmscordaccore.dylib
 Referenced from: /private/tmp/helix/working/9BD708E3/p/createdump
 Reason: image not found
/private/tmp/helix/working/9BD708E3/w/A0560885/e/baseservices/threading/generics/TimerCallback/thread21/thread21.sh: line 377: 46534 Abort trap: 6 (core dumped) $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"

Return code: 1
Raw output file: /private/tmp/helix/working/9BD708E3/w/A0560885/e/baseservices/threading/Reports/baseservices.threading/generics/TimerCallback/thread21/thread21.output.txt
Raw output:
BEGIN EXECUTION
/tmp/helix/working/9BD708E3/p/corerun thread21.dll ''
Expected: 100
Actual: 134
END EXECUTION - FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=/tmp/helix/working/9BD708E3/p
> /private/tmp/helix/working/9BD708E3/w/A0560885/e/baseservices/threading/generics/TimerCallback/thread21/thread21.sh
Expected: True
Actual: False

Stack trace
   at baseservices_threading._generics_TimerCallback_thread21_thread21_._generics_TimerCallback_thread21_thread21_sh()

```

**Comment by BruceForstall** [comment]
@sdmaclea @sandreenko OSX arm64 GCStress + JitStress failures

**Comment by BruceForstall** [comment]
Another failure:

https://dev.azure.com/dnceng/public/_build/results?buildId=1082595&view=ms.vss-test-web.build-test-results-tab&runId=33198112&resultId=104873&paneView=debug

seems very consistent. Only jitstress1 + GCStress=C

**Comment by sdmaclea** [comment]
We are waiting on the macOS 11.3 release which seems to fix most of these transient issues.

If you are willing to disable, you can disable any failing Apple Silicon test on first failure as long as we have an issue open to track the disabled test(s).

**Comment by mangod9** [comment]
These are fixed after 11.3 I believe.
